File: py_solutions_1-10/Euler_1.py
## sum of all multiples of 3 and 5 below 1000? ##

import timeit


start = timeit.default_timer()

def euler_1():
    return sum([i for i in xrange(1, 1000) if i % 3 == 0 or i % 5 == 0])

# with reduce
#def euler_1():
#    add = lambda x, y: x + y
#    return reduce(add, [i for i in xrange(1000) if i % 3 == 0 or i % 5 == 0])

# with filter
#def euler_1():
#    three_or_five = lambda x: x % 3 == 0 or x % 5 == 0
#    return sum(filter(three_or_five, xrange(1000)))

print "Answer: %s" % euler_1()
stop = timeit.default_timer()
print "Time: %f" % (stop - start)

File: java_solutions_1-10/Euler_1.java
/* 
/   Project Euler problem #1 Multiples of 3 and 5
/
/   Find the sum of all the multiples of 3 or 5 below 1000
/
*/


public class Euler_1
{
    public static int factors(int limit, int ... factorsOf)
    {
        int sum = 0;
        int numFactors = factorsOf.length;

        for (int j, i = 0; i < limit; i++) {
            for (j = 0; j < numFactors; j++)
                if (i % factorsOf[j] == 0) break;
            sum += j == numFactors ? 0 : i;
        }

        return sum;
    }           

    public static void main(String[] args)
    {
        long start = System.nanoTime();

        int factor_sum = factors(1000, 3, 5);

        long stop = System.nanoTime();
        System.out.println("Answer: " + factor_sum); 
        System.out.printf("Time: %.7f\n", ((float) stop - (float) start) / 10000000);

    }
}    
